src/UsgsAstroSarSensorModel.cpp

```diff
@@ -136,3 +141,209 @@ string UsgsAstroSarSensorModel::constructStateFromIsd(
   // some state data is not in the ISD and requires a SM to compute them.
   return state.dump();
 }
+
+// TEMPORARY hard-coded values. Need to be replaced by accessors.
+double START_TIME = 0.0;
+double STOP_TIME = 5.0;                         
+double EXPOSURE_DURATION = 0.005;
+double SCALED_PIXEL_WIDTH = 7.5; 
+double WAVELENGTH = 0.125;
+double SAMPLES = 1000.0;
+double LINES = SAMPLES;
+double groundRangeFunc(double x);
+
+// Needs to be a function that can interpolate between existing times
+double a1 = 7.99423808710000e+04;
+double a2 = 6.92122900000000e-01;
+double a3 = 3.40193700000000e-06;
+double a4 = -2.39924200000000e-11;
+double slantRange = 0.12;
+
+double secantRoot(std::function<double(double)> lambda, double x1, double xh, 
+                  double tolerance, double maxIters);
+
+csm::ImageCoord UsgsAstroSarSensorModel::groundToImage(
+    const csm::EcefCoord& groundPt,
+    double desiredPrecision,
+    double* achievedPrecision,
+    csm::WarningList* warnings) const {
+
+  //MESSAGE_LOG(this->m_logger, "Computing groundToImage(ImageCoord) for {}, {}, {}, with desired precision {}",
+  //          groundPt.x, groundPt.y, groundPt.z, desiredPrecision);
+
+  // Find time of closest approach to groundPt and the corresponding slant range by finding 
+  // the root of the doppler shift frequency
+  csm::EcefCoord newGroundPt(groundPt);
+  double time = dopplerShiftRoot(newGroundPt);
+  double slantRange = calcSlantRange(newGroundPt, time);
+
+  if (time != -1) {
+    // Find the ground range, based on the ground-range-to-slant-range polynomial defined by the 
+    // range coefficient set, with a time closest to the calculated time of closest approach
+    double groundRange = slantToGroundRange(groundPt, time, slantRange);
+
+    if (groundRange =! -1) {
+      double line = 1 + (time - START_TIME) / EXPOSURE_DURATION;
+      double sample = 1 + groundRange / SCALED_PIXEL_WIDTH;
```

Comment:
Socet Set and CSM use different coordinate systems. I think the 1+ needs to go away in each of these.

Reply:
You're right, thanks for catching this.

Reply:
Are ISIS and CSM 1 off or 0.5 off?